src/writer.rs

```diff
@@ -85,6 +91,28 @@ impl<'a, D: Distance, R: Rng + SeedableRng> ArroyBuilder<'a, D, R> {
         self
     }
 
+    /// Configure the maximum memory arroy can use to build its trees in bytes.
+    ///
+    /// This value is used as a hint; arroy may still consume too much memory, especially if the value is too low.
+    /// If not specified, arroy will use as much memory as possible but keep in mind that if arroy tries to use more
+    /// memory than you have, it'll become very slow.
```

Comment:
```suggestion
    /// memory than you have, it'll become very slow.
    ///
    /// In this case, it will randomly read the disk as pages will be invalidated by other reads, and OS cache will
    /// become useless.
```